### PR TITLE
limit Android NDK jobs on CI

### PR DESCRIPTION
**Pull request by dulmandakh**
Limit number of NDK jobs of ReactAndroid on CI using $BUILD_THREADS environment variable. Otherwise, it was spawning 32 jobs while building RNTester, which caused in OOM or unexpected failure.

CI: https://circleci.com/gh/dulmandakh/react-native/322

## Test Plan

Android CI is green again 😍

### CODE REVIEW

**Comment by gengjiawen** [comment]
aHa, That's why. My first thought is the gradle need more memeory.

File: ReactAndroid/build.gradle [hunk 1]
```diff
@@ -221,7 +222,7 @@ task buildReactNdkLib(dependsOn: [prepareJSC, prepareBoost, prepareDoubleConvers
             "THIRD_PARTY_NDK_DIR=$buildDir/third-party-ndk",
             "REACT_COMMON_DIR=$projectDir/../ReactCommon",
             '-C', file('src/main/jni/react/jni').absolutePath,
-            '--jobs', project.hasProperty("jobs") ? project.property("jobs") : Runtime.runtime.availableProcessors()
+            '--jobs', ndkJobs ?: project.hasProperty("jobs") ? project.property("jobs") : Runtime.runtime.availableProcessors()
```
Comment (gengjiawen) [inline ReactAndroid/build.gradle]: With two ?: this looks a little confusing, maybe split it up. Are you revert to jobs if ndkJobs not defined ?

**Comment by dulmandakh** [comment]
@hramos Please review and merge 😉

**Comment by dulmandakh** [comment]
cleaner version https://github.com/facebook/react-native/pull/19755

**Pull request closed without merging**